backend/src/main/kotlin/com/group7/artshare/entity/DiscussionPost.kt

```diff
@@ -57,8 +70,12 @@ class DiscussionPost {
         discussionPostDTO.textBody = this.textBody
         discussionPostDTO.creationDate = this.creationDate
         discussionPostDTO.lastEditDate = this.lastEditDate
-        discussionPostDTO.upvoteNo = this.upvoteNo
-        discussionPostDTO.downvoteNo = this.downvoteNo
+        for(user in this.upVotedUsers){
+            discussionPostDTO.upVotedUserIds.add(user.id)
```

Comment:
you can use map instead of for. something like:
discussionPostDTO.upVotedUserIds = this.upVotedUsers.map{it.id} or
discussionPostDTO.upVotedUsernames = this.upVotedUsers.map{it.accountInfo.username}

Reply:
Thanks for letting me know this kind of usage exists. I will watchout for it in future.